borgbackup: Improve documentation

ngiger opened the pull request:
Added an example on howto create a backup job to
a remote borg server.

Tested with my borgbase.com account.

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/manual-does-not-specify-who-to-cc-when-submitting-a-patch/6449/1

rnhmjoj commented:
Thank you for contributing! 
I think an example with comments and this long would be better suited for the NixOS manual rathen than the option help. It would be really helpful if you could write a (even small) section on bogbackup.
In case here's a [guide](https://nixos.org/nixos/manual/index.html#sec-writing-documentation) on how to edit the manual.

ngiger committed:
borgbackup: Moved documentation to NixOS manual and added examples for
* creating a local backup
* creating a borgbackup server
* backing up to a borgbackup server
* hints about the Vorta graphical desktop application
* Added documentation about Vorta desktop client

Tested the examples locally and with my borgbase.com account.

ngiger commented:
Thanks for your suggestion. Please review also my second patch.
As it is my first contribuion I might well miss some common practices for NixOS, but I invested a few hours and looked at the documentation of postgresql and nextcloud and tried to pick the best out of it.
Also it would be nice if @dotlambda could review these changes, too.

rnhmjoj commented on nixos/modules/services/backup/borgbackup.nix:
```diff
@@ -197,10 +198,12 @@ in {
       Deduplicating backups using BorgBackup.
       Adding a job will cause a borg-job-NAME wrapper to be added
       to your system path, so that you can perform maintenance easily.
+      See also the chapter about  See also the chapter about BorgBackup in the NixOS manual.
```
Typo?

rnhmjoj commented on nixos/modules/services/backup/borgbackup.xml:
```diff
@@ -0,0 +1,180 @@
+          doInit = true;
+          encryption = {
+            mode = "repokey";
+            passphrase = "secret";
```
Can you warn about the `passphrase` being world-readable and recommend using `passCommand`?

rnhmjoj commented on nixos/modules/services/backup/borgbackup.xml:
```diff
@@ -0,0 +1,180 @@
+  <link xlink:href="https://borgbackup.readthedocs.io/"/>
+ </para>
+ <para>
+  <link xlink:href="https://www.borgbackup.org/">BorgBackup</link> (short: Borg) is a deduplicating backup program. Optionally, it supports compression and authenticated encryption.
```
Can you try to keep the lines under 80 characters?

rnhmjoj reviewed (changes requested):
Wow, excellent work! I left a few comments, besides that it's good to be merged.
Thank you for your work.

ngiger commented:
Fixed the typo.
Limited the length of documentation to 80 chars (except a few lines which are output
of bash shell.
Added the warning about the passphrase and used passCommand in all other examples.

Mic92 commented on nixos/modules/services/backup/borgbackup.xml:
```diff
@@ -0,0 +1,209 @@
+    test as of end of March 20 failed with the error
+<programlisting>
+Error: Error deploying: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown:
+The name org.freedesktop.Accounts was not provided by any .service files</programlisting>
```
Untested but does enabling the following helps: `services.accounts-daemon.enable = true;` ?

ngiger replied:
Thanks for tip. This solves the problem. But I think this warrants a different issues/pull request and I do not know how to fix this kind of problem

rnhmjoj replied:
This is a general problem: Nix packages don't have a way to specify a dependency on a module, service or really anything provided by NixOS. A similar issue was reported here: #79075

ngiger replied:
Okay. In this case I will add your tip to the manual. Just wait a little bit, I have to leave office for a while.

ngiger replied:
I pushed the last commit. I think it should be okay now to pull this request.
Or are there still other problems?

rnhmjoj commented on nixos/modules/services/backup/borgbackup.xml:
```diff
@@ -0,0 +1,227 @@
+ <title>BorgBackup</title>
+  <para>
+  <emphasis>Source:</emphasis>
+  <filename>modules/services/backup/rsnapshot.nix</filename>
```
```suggestion
  <filename>modules/services/backup/borgbackup.nix</filename>
```

Found a mistake here.

rnhmjoj replied:
I don't see any other problem. Once you are done, please squash the commits  (I use `git rebase -i` for this) and I'll merge the PR.

ngiger replied:
Okay. Rebase done. The `git rebase -i `is now much more user friendly. I stemmed away of using it a few years ago, but this time it worked like a charm. Thanks for the tip.

timokau commented:
This broke eval on master because of the `''` in the comment being interpreted as the end of the literal example.

timokau commented:
https://github.com/NixOS/nixpkgs/pull/84102

timokau commented:
What do you mean? It seemed to work fine before:  https://github.com/NixOS/nixpkgs/pull/84096

Mic92 commented:
@timokau Ok. It was broken yesterday. But looks like it got fixed again.

rnhmjoj commented:
Sorry I wasn't around at the time... 
I just want to thank @ngiger again for this great contribution. NixOS really need more documentation like this.
